src/System.CommandLine.Hosting/HostingExtensions.cs

```diff
@@ -141,11 +141,14 @@ public static InvocationContext GetInvocationContext(this HostBuilderContext con
             throw new InvalidOperationException("Host builder has no Invocation Context registered to it.");
         }
 
-        public static IHost GetHost(this InvocationContext invocationContext)
+        public static IHost? GetHost(this InvocationContext invocationContext)
         {
             _ = invocationContext ?? throw new ArgumentNullException(paramName: nameof(invocationContext));
             var hostModelBinder = new ModelBinder<IHost>();
-            return (IHost)hostModelBinder.CreateInstance(invocationContext.BindingContext);
+            var instance = hostModelBinder.CreateInstance(invocationContext.BindingContext);
+            if (instance is IHost host)
+              return host;            
+            return null;
```

Comment:
More succinct syntax could be (assuming we keep the null return):
```suggestion
            return hostModelBinder.CreateInstance(invocationContext.BindingContext) as IHost;
```